src/hardhat.rs

```diff
@@ -0,0 +1,101 @@
+use std::sync::{Arc, RwLock};
+
+use crate::{fork::ForkSource, node::InMemoryNodeInner};
+use jsonrpc_core::{BoxFuture, Result};
+use jsonrpc_derive::rpc;
+use zksync_basic_types::{AccountTreeId, Address, U256};
+use zksync_core::api_server::web3::backend_jsonrpc::error::into_jsrpc_error;
+use zksync_types::{utils::storage_key_for_standard_token_balance, L2_ETH_TOKEN_ADDRESS};
+use zksync_utils::u256_to_h256;
+use zksync_web3_decl::error::Web3Error;
+
+/// Implementation of HardhatNamespaceImpl
+pub struct HardhatNamespaceImpl<S> {
+    node: Arc<RwLock<InMemoryNodeInner<S>>>,
+}
+
+impl<S> HardhatNamespaceImpl<S> {
+    /// Creates a new `Hardhat` instance with the given `node`.
+    pub fn new(node: Arc<RwLock<InMemoryNodeInner<S>>>) -> Self {
+        Self { node }
+    }
+}
+
+#[rpc]
+pub trait HardhatNamespaceT {
+    #[rpc(name = "hardhat_setBalance")]
```

Comment:
Please update the `SUPPORTED_APIS.md` with the updated information/example of this endpoint

Reply:
done